### DISCUSSION TITLE
support cloud event

### DISCUSSION THREAD

**Discussion by tangxusc** [apache/pulsar#18827] [General]
**Is your feature request related to a problem? Please describe.**
Does pulsar now support cloud event for cncf? Are there any plans for future support?
Is there a support plan for knative event components?

**Describe the solution you'd like**
support cloudevents and knative

**Describe alternatives you've considered**
support https://cloudevents.io/ CloudEvents 0.3
knative now support kafka,It is hoped that pulsar can also be accessed

**Additional context**
none

**Comment by aahmed-se** [comment]
It's been discussed though not enough requests have been made to give consideration.

**Reply by graveviolin** [reply]
How many requests is needed to give this consideration again? I see 25 thumbs up.

**Reply by tisonkun** [reply]
See https://github.com/apache/pulsar/discussions/18827#discussioncomment-4348975 @graveviolin there is no "product team" in an ASF project, so you'd be the volunteer or find volunteers to work on it.

No blocker for this faeture, just no volunteers to work on it.

**Reply by graveviolin** [reply]
Thanks! I saw a bit more after I posted. Thanks for context, let's see if I can contribute something.

**Comment by j4fm** [comment]
+1

**Comment by sijie** [comment]
@tangxusc there was actually a proposal I sent to cloud events a year ago. we can pick it up again. also if you are interested in contributing this, you are very welcome to do so.

**Comment by aslom** [comment]
@sijie how can I find that proposal? link?

**Comment by sijie** [comment]
for people who are interested in this topic, here is the proposal : https://github.com/cloudevents/spec/pull/237/files

**Comment by sijie** [comment]
The proposal was a bit old. I think the CloudEvents has evolved to 1.0. So the proposal might need to be reworked to 1.0 specification.

**Comment by johanhaleby** [comment]
I think this would be a great contribution.

**Comment by devinbost** [comment]
I'm interested in this. We're taking a serious look at CloudEvents.

**Comment by tisonkun** [comment]
Moved to the Discussion forum since there's no design or volunteer works on it now.

It requires a PIP to add such a support.
